File: 1376-time-needed-to-inform-all-employees/1376-time-needed-to-inform-all-employees.cpp
class Solution {
public:

    int time(unordered_map<int, vector<int>>& children, vector<int>& informTime, int node) {
        int largest = 0;
        for (int child : children[node]) {
            largest = max(largest, time(children, informTime, child));
        }
        return informTime[node] + largest;
    }

    int numOfMinutes(int n, int headID, vector<int>& manager, vector<int>& informTime) {
        
        unordered_map<int, vector<int>> children;
        for (int i = 0; i < n; i++) {
            children[manager[i]].push_back(i);
        }

        return time(children, informTime, headID);
        
    }
};

File: 1376-time-needed-to-inform-all-employees/1376-time-needed-to-inform-all-employees.py
class Solution:
    def numOfMinutes(self, n: int, headID: int, manager: List[int], informTime: List[int]) -> int:
        
        children = {}
        for i, parent in enumerate(manager):
            if parent not in children:
                children[parent] = []
            children[parent].append(i)
        
        def time(node):
            largest = 0
            for child in children.get(node, []):
                largest = max(largest, time(child))
            return informTime[node] + largest
        
        return time(headID)
